py/dml/dmlc.py

```diff
@@ -624,6 +621,11 @@ def main(argv):
         prerr("dmlc: the -P flag cannot be used together with -g")
         sys.exit(1)
 
+    if options.dep:
+        for tag in dir(messages):
+            if is_warning_tag(tag):
+                ignore_warning(tag)
```

Comment:
Tried it, I think I like it somewhat better (although the use of a global is also somewhat brutal)

Reply:
Actually, I just realized, the definition should be:
```
def warning_is_ignored(tag):
    return dml.globals.ignore_all_warnings or ignored_warnings.get(tag, False)
```
Otherwise, explicitly enabled warnings will still get warned about with `--dep` on.

Reply:
Feeling stupid. Fixed!